astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -244,32 +243,38 @@ def precessedgeo_to_gcrs(from_coo, to_frame):
     pmat = gcrs_precession_mat(from_coo.equinox)
     crepr = from_coo.cartesian.transform(matrix_transpose(pmat))
     gcrs_coo = GCRS(crepr,
-                    obstime=to_frame.obstime,
-                    obsgeoloc=to_frame.obsgeoloc,
-                    obsgeovel=to_frame.obsgeovel)
+                    obstime=from_coo.obstime,
```

Comment:
This change suggests we had a bug (your code looks correct, but checking). Is it worth splitting this out as a separate bugfix? Is this something we can write a test case for? Would be good to have a changelog entry as well...

Reply:
Now extracted as #12152, along with the bugfix for the `TEME` transformations;  I've also added a test there.